docs/guides/data-movement/copy-offload.md

```diff
@@ -0,0 +1,92 @@
+# Copy-Offload
+
+The copy-offload API allows a user's compute application to specify [Data Movement](../data-movement/readme.md) requests. The user's application utilizes the `libcopyoffload` library to establish a secure connection to the copy-offload server to initiate, list, query the status of, or cancel data movement requests. The copy-offload server accepts only those requests that present that Workflow's token.
+
+The copy-offload server is implemented as a special kind of [User Container](../user-containers/readme.md). Like all user containers, this is activated by a `DW container` directive in the user's job script and runs on the Rabbit nodes that are associated with the compute nodes in the user's job.
+
+## Administrative Configuration
+
+### TLS signing key and certificate
+
+A signing key and self-signed TLS certificate must be created and made available to the copy-offload server and the certificate must also be copied to each compute node. This certificate must have a SAN extension that describes all of the Rabbit nodes.
+
+Tools are available in the `nnf-dm` repository to assist in creating this certificate and its signing key. Begin by confirming that the cluster's `SystemConfiguration` resource can be accessed using the 'kubectl' command. This resource contains the information about all of the Rabbit nodes and is used when creating the SAN extension for the certificate:
+
+```console
+kubectl get systemconfiguration
+```
+
+From the `nnf-deploy` workarea, change into the `nnf-dm` directory and run `tools/mk-usercontainer-secrets.sh`:
+
+```console
+cd nnf-dm
+tools/mk-usercontainer-secrets.sh
+```
+
+That tool creates the signing key and the certificate and stores them in a Kubernetes secret named `nnf-dm-usercontainer-server-tls`. This first secret is mounted into the copy-offload server's pod when it is specified in a user's job script. The certificate is also stored alone in a Kubernetes secret named `nnf-dm-usercontainer-client-tls`. The content of this second secret can be retrieved by the administrator and copied to each compute node.
```

Comment:
> stored alone

What do you mean by this?

Reply:
The first secret has both the cert and its key. The second secret has only the cert--alone. I'll look at re-phrasing this.